rewrite-java/src/main/java/org/openrewrite/java/AddLiteralMethodArgument.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.apache.commons.lang3.StringUtils;
+import org.openrewrite.*;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.openrewrite.Tree.randomId;
+import static org.openrewrite.java.tree.JavaType.Primitive;
+
+/**
+ * This recipe finds method invocations matching a method pattern and uses a zero-based argument index to determine
+ * which argument is added with a literal value.
+ */
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddLiteralMethodArgument extends Recipe {
+
+    /**
+     * A method pattern that is used to find matching method invocations.
+     * See {@link  MethodMatcher} for details on the expression's syntax.
+     */
+    @Option(displayName = "Method pattern",
+            description = "A method pattern that is used to find matching method invocations.",
+            example = "com.yourorg.A foo(int, int)")
+    String methodPattern;
+
+    /**
+     * A zero-based index that indicates which argument will be added as null to the method invocation.
+     */
+    @Option(displayName = "Argument index",
+            description = "A zero-based index that indicates which argument will be added as null to the method invocation.",
+            example = "0")
+    int argumentIndex;
+
+    @Option(displayName = "Literal",
+            description = "The literal value that we add the argument for.",
+            example = "0")
```

Comment:
Since the type example below is a String, and to differentiate this from the index, I thought it best to use a string value here too.
```suggestion
            example = "abc")
```